app/components/app-monitoring/settings/index.ts

```diff
@@ -2,22 +2,62 @@ import { inject as service } from '@ember/service';
 import Component from '@glimmer/component';
 import { task } from 'ember-concurrency';
 import IntlService from 'ember-intl/services/intl';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import RouterService from '@ember/routing/router-service';
+import { debounce } from '@ember/runloop';
+
 import AmConfigurationModel from 'irene/models/amconfiguration';
 import MeService from 'irene/services/me';
 import OrganizationService from 'irene/services/organization';
+import styles from './index.scss';
 import AppMonitoringService from 'irene/services/appmonitoring';
+import ENUMS from 'irene/enums';
 
 interface AppMonitoringSettingsSignature {
   Args: {
     settings: AmConfigurationModel | undefined;
   };
 }
 
+interface PlatformObject {
+  key: string;
+  value: number;
+}
+
 export default class AppMonitoringSettingsComponent extends Component<AppMonitoringSettingsSignature> {
   @service declare intl: IntlService;
   @service declare me: MeService;
   @service declare organization: OrganizationService;
   @service declare appmonitoring: AppMonitoringService;
+  @service declare router: RouterService;
+
+  @tracked selectedPlatform = {
+    key: 'All',
+    value: -1,
+  };
+
+  @tracked searchQuery = '';
+  @tracked platform = -1;
+
+  constructor(owner: unknown, args: AppMonitoringSettingsSignature['Args']) {
+    super(owner, args);
+
+    const queryParamObj = this.router.currentRoute?.queryParams;
+
+    if (queryParamObj && queryParamObj['app_platform']) {
+      for (const platformObject of this.platformObjects) {
+        if (platformObject.value === Number(queryParamObj['app_platform'])) {
+          this.selectedPlatform = platformObject;
+          break;
+        }
+      }
+    }
+
+    if (queryParamObj && queryParamObj['app_query']) {
+      this.searchQuery = queryParamObj['app_query'];
+    }
+  }
```

Comment:
we can change `selectedPlatform` and `searchQuery` to a getter to simplify this logic

```
get selectedPlatform() {
  const queryParams = this.router.currentRoute?.queryParams;

  return this.platformObjects.find(po => po.value === (queryParams ? Number(queryParams['app_platform']) : -1)) || {
    key: 'All',
    value: -1,
  }
}
```
and similiar getter `searchQuery` too